python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -326,6 +327,7 @@ def to_metrics_dict(self) -> Dict[str, int]:
 
 def _map_task(
     fn: Callable[[Iterator[Block]], Iterator[Block]],
+    ctx: TaskContext,
```

Comment:
`fn` type needs to be updated as well.

`MapOperator.__init__(transform_fn)` type also needs to be updated.